mod/state-transition/pkg/core/state_processor_genesis.go

```diff
@@ -102,7 +97,18 @@ func (sp *StateProcessor[
 	}
 
 	for _, deposit := range deposits {
-		if err := sp.processDeposit(st, deposit); err != nil {
+		prevDepositIndex, err := st.GetEth1DepositIndex()
+		if err != nil {
+			// currently this only happen on the first genesis deposit
+			if deposit.GetIndex() != 0 {
+				return nil, ErrDepositMismatch
+			}
```

Comment:
should we ignore the err if its non-nil `&& deposit.GetIndex() == 0` ?

Reply:
should still return err imo

Reply:
For the very first deposit, where `Eth1DepositIndex` has not been set yet, we have a ErrNotFound kind of error. This is fine (we haven't even set a single deposit) but it should only happen for the very first deposit (so the check on deposit index)
On the other hand, I agree that we should check the kind of error we receive, and we should behave like that only for ErrNotFound and not others but:
- ErrNotFound is a cosmosSDK error and we don't want that dependency here (discussed with Dev at some point)
- We do this in another place (see `applyDeposit` and how we do create validators
- I have an old PR that start fixing it (https://github.com/berachain/beacon-kit/pull/2065) and I am willing to refresh that and resubmit for review as soon as this in in